### DISCUSSION TITLE
[ZONE] in template

### DISCUSSION THREAD

**Discussion by ITC-KEDA** [ngoduykhanh/PowerDNS-Admin#1654] [Q&A] [answered]
I have try to use [ZONE] in my tempate
[https://files.itconfig.dk/download/upload/2023-08-24%2016_26_51-Edit%20Template%20-%20IT%20Confidence%20DNS%20Admin.png](url)

But it not woring properly, as when create a zone with that template it just create a empty domain, with just the SOA entry.
I have look into my /app/powerdnsadmin/models/record.py file
and it are including: 
record['record_data'] = record['record_data'].replace('[ZONE]', domain_name)

Can anyone help, what iam during wrong?

**Comment by ITC-KEDA** [comment]
Just tested it. It working inside a domain, so if i created the domain, and go into it and create an record. I can type [ZONE] into record data, an after Apply, it have type in the real domain name.

But it not working in template?

**Comment by ITC-KEDA** [comment]
Could anyone create a patch for this?

**Comment by AzorianMatt** [comment, marked as answer]
@ITC-KEDA I ran a test with the current release `0.4.2` and was not able to replicate this problem. Here are my steps:

1. Create new blank template.
2. Create CNAME record in new template with the content "[ZONE]" and save.
3. Create new zone from the previously created template.
4. New zone has CNAME record with content matching the new zone's apex.

Please make sure you are using the appropriate image as follows: `powerdnsadmin/pda-legacy:0.4.2`

**Comment by ITC-KEDA** [comment]
It works with a new template, maybe som old legacy stuff in the other templates.
Thanks you for returing to me with this!.

**Reply by AzorianMatt** [reply]
You're welcome.
